client/components/search/index.jsx

```diff
@@ -147,19 +154,13 @@ class Search extends Component {
 		if ( this.state.keyword === prevState.keyword ) {
 			return;
 		}
-		// if there's a keyword change: trigger search
-		if ( this.state.keyword ) {
-			// this.onSearch is debounced when this.props.delaySearch === true
-			// this avoids unnecessary fetches while user types
-			this.onSearch( this.state.keyword );
-		} else {
-			// this.props.onSearch is _not_ debounced
-			// no need to debounce if ! this.state.keyword
-			if ( this.props.delaySearch ) {
-				// Cancel any pending debounce
-				this.onSearch.cancel();
-			}
-			this.props.onSearch( this.state.keyword );
+		// if there's a keyword change and mode is search-on-write: trigger search
+		// The empty string must be handled always becuase it can be triggered:
+		// - by the user clicking on the clear icon
+		// - by the user pressing the ESC key
+		// Additionally, the search can be triggered on demand by using the boolean prop `applySearch`
+		if ( this.isSearchOnWriteMode() || this.state.keyword === '' || this.props.applySearch ) {
+			this.updateSearch();
```

Comment:
I'm a bit worried about this check, because it doesn't verify if the keyword was changed and only if the keyword exists. What could happen here is the search being refired sometimes due to a change on other properties, right?

But I also understand this behavior was not added on your PR.

Reply:
That verification of the keyword being changed is done in the previous statement so that scenario is not really a problem here 😌

Reply:
That is right, it was not showing on the diff and I didn't check before, my bad 🤐